src/liballoc/allocator.rs

```diff
@@ -464,6 +464,27 @@ impl fmt::Display for CannotReallocInPlace {
 ///  * if a layout `k` fits a memory block (denoted by `ptr`)
 ///    currently allocated via an allocator `a`, then it is legal to
 ///    use that layout to deallocate it, i.e. `a.dealloc(ptr, k);`.
+///
+/// # Unsafety
+///
+/// The `Alloc` trait is an `unsafe` trait for a number of reasons, and
+/// implementors must ensure that they adhere to these contracts:
+///
+/// * Pointers returned from allocation functions must point to valid memory and
+///   retain their validity until at least the instance of `Alloc` is dropped
+///   itself.
+///
+/// * Implementations of `Alloc` are not currently memory safe if they unwind.
```

Comment:
Wait, I'm not clear on how to parse this. The way it reads now, I interpret this as saying that *any* implementation of `Alloc` is not memory safe if it has a method that unwinds (panics), which AFAICT would even include the `fn oom` method...

But if I'm understanding the change here correctly, the memory unsafety only arises for implementations that are installed as *global allocators*, right?

(Maybe we do need to make a `unsafe trait GlobalAllocator: Allocator { }` just to track this extra-constraint on the implementations then...?)